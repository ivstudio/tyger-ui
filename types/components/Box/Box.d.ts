/// <reference types="react" />
import { TSpacing } from '../../utils/spacing';
export declare type TBoxElem = 'div' | 'section' | 'span' | 'article' | 'main' | 'ul' | 'ol' | 'li' | 'aside';
export declare type TBox = {
    pt?: TSpacing;
    pr?: TSpacing;
    pb?: TSpacing;
    pl?: TSpacing;
    py?: TSpacing;
    px?: TSpacing;
    mt?: TSpacing;
    mr?: TSpacing;
    mb?: TSpacing;
    ml?: TSpacing;
    my?: TSpacing;
    mx?: TSpacing;
    p?: TSpacing;
    m?: TSpacing;
    border?: boolean;
    borderRadius?: boolean;
};
export interface IBox extends TBox {
    children: React.ReactNode | React.ReactNode[];
    tag?: TBoxElem;
    flexCenterAll?: boolean;
    flexCenterInline?: boolean;
    flex?: boolean;
    styles?: React.CSSProperties;
}
export declare const BoxRoot: import("styled-components").StyledComponent<"div", import("styled-components").DefaultTheme, IBox, never>;
declare const Box: ({ tag, children, styles, ...props }: IBox) => JSX.Element;
export default Box;
