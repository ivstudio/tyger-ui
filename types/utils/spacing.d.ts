export declare const spacing: {
    readonly '0': "0px";
    readonly '2': "2px";
    readonly '4': "4px";
    readonly '8': "8px";
    readonly '12': "12px";
    readonly '16': "16px";
    readonly '24': "24px";
    readonly '32': "32px";
    readonly '40': "40px";
    readonly '48': "48px";
    readonly '64': "64px";
    readonly '80': "80px";
    readonly '96': "96px";
    readonly '160': "160px";
};
export declare type TSpacing = keyof typeof spacing;
