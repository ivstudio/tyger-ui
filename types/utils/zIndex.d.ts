export declare const zIndex: {
    readonly mobileStepper: 1000;
    readonly appBar: 1100;
    readonly drawer: 1200;
    readonly modal: 1300;
    readonly snackbar: 1400;
    readonly tooltip: 1500;
};
export declare type TZIndex = keyof typeof zIndex;
