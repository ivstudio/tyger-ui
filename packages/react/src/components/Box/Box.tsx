'use client';

import styled from 'styled-components';

import {
    flexCenterAll as flexCenterAllMixin,
    flexCenterInline as flexCenterInlineMixin,
} from '../../styles';
import { BoxProps } from './Box.types';

export const BoxRoot = styled.div<BoxProps>`
    padding-top: ${({ theme, pt, py }) =>
        (py && theme.spacing[py]) || (pt && theme.spacing[pt])};
    padding-right: ${({ theme, pr, px }) =>
        (px && theme.spacing[px]) || (pr && theme.spacing[pr])};
    padding-bottom: ${({ theme, pb, py }) =>
        (py && theme.spacing[py]) || (pb && theme.spacing[pb])};
    padding-left: ${({ theme, pl, px }) =>
        (px && theme.spacing[px]) || (pl && theme.spacing[pl])};
    margin-top: ${({ theme, mt, my }) =>
        (my && theme.spacing[my]) || (mt && theme.spacing[mt])};
    margin-right: ${({ theme, mr, mx }) =>
        (mx && theme.spacing[mx]) || (mr && theme.spacing[mr])};
    margin-bottom: ${({ theme, mb, my }) =>
        (my && theme.spacing[my]) || (mb && theme.spacing[mb])};
    margin-left: ${({ theme, ml, mx }) =>
        (mx && theme.spacing[mx]) || (ml && theme.spacing[ml])};
    display: ${({ flex }) => flex && 'flex'};
    ${({ flexCenterInline }) => flexCenterInline && flexCenterInlineMixin};
    ${({ flexCenterAll }) => flexCenterAll && flexCenterAllMixin}
    padding: ${({ theme, padding }) => padding && theme.spacing[padding]};
    margin: ${({ theme, margin }) => margin && theme.spacing[margin]};
    border: ${({ theme, border }) => border && `1px solid ${theme.border}`};
    border-radius: ${({ borderRadius }) => borderRadius && '4px'};
`;

const Box = ({ tag, children, styles, ...props }: BoxProps) => (
    <BoxRoot as={tag} style={styles} {...props}>
        {children}
    </BoxRoot>
);

export default Box;
