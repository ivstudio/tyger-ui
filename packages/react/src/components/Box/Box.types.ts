import type { SpacingProps } from '@tyger-ui/system';

export type BoxTags =
    | 'div'
    | 'p'
    | 'section'
    | 'span'
    | 'article'
    | 'main'
    | 'ul'
    | 'ol'
    | 'li'
    | 'aside'
    | 'fieldset';

export type BoxRootProps = {
    pt?: SpacingProps;
    pr?: SpacingProps;
    pb?: SpacingProps;
    pl?: SpacingProps;
    py?: SpacingProps;
    px?: SpacingProps;
    mt?: SpacingProps;
    mr?: SpacingProps;
    mb?: SpacingProps;
    ml?: SpacingProps;
    my?: SpacingProps;
    mx?: SpacingProps;
    padding?: SpacingProps;
    margin?: SpacingProps;
    border?: boolean;
    borderRadius?: boolean;
};

export interface BoxProps extends BoxRootProps {
    children: React.ReactNode | React.ReactNode[];
    tag?: BoxTags;
    flexCenterAll?: boolean;
    flexCenterInline?: boolean;
    flex?: boolean;
    styles?: React.CSSProperties;
}
