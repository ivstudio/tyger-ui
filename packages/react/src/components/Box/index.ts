'use client';
export * from './Box';
export { default } from './Box';
