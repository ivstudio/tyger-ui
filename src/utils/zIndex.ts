export const zindex = {
    mobileStepper: 1000,
    appBar: 1100,
    drawer: 1200,
    modal: 1300,
    snackbar: 1400,
    tooltip: 1500,
};
export type TZIndex = keyof typeof zindex;
