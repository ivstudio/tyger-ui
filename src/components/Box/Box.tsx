import styled from 'styled-components';

import {
    flexCenterAll as flexCenterAllMixin,
    flexCenterInline as flexCenterInlineMixin,
} from '../../styles';
import { TSpacing } from '../../utils/spacing';

export type TBoxElem =
    | 'div'
    | 'section'
    | 'span'
    | 'article'
    | 'main'
    | 'ul'
    | 'ol'
    | 'li'
    | 'aside';

export type TBox = {
    pt?: TSpacing;
    pr?: TSpacing;
    pb?: TSpacing;
    pl?: TSpacing;
    py?: TSpacing;
    px?: TSpacing;
    mt?: TSpacing;
    mr?: TSpacing;
    mb?: TSpacing;
    ml?: TSpacing;
    my?: TSpacing;
    mx?: TSpacing;
};

export interface IBox extends TBox {
    children: React.ReactNode | React.ReactNode[];
    tag?: TBoxElem;
    flexCenterAll?: boolean;
    flexCenterInline?: boolean;
}

export const BoxRoot = styled.div<IBox>`
    padding-top: ${({ theme, pt, py }) =>
        py || (pt && theme.spacing[py || pt])};
    padding-right: ${({ theme, pr, px }) =>
        px || (pr && theme.spacing[px || pr])};
    padding-bottom: ${({ theme, pb, py }) =>
        py || (pb && theme.spacing[py || pb])};
    padding-left: ${({ theme, pl, px }) =>
        px || (pl && theme.spacing[px || pl])};
    margin-top: ${({ theme, mt, my }) => my || (mt && theme.spacing[my || mt])};
    margin-right: ${({ theme, mr, mx }) =>
        mx || (mr && theme.spacing[mx || mr])};
    margin-bottom: ${({ theme, mb, my }) =>
        my || (mb && theme.spacing[my || mb])};
    margin-left: ${({ theme, ml, mx }) =>
        mx || (ml && theme.spacing[mx || ml])};
    ${({ flexCenterInline }) => flexCenterInline && flexCenterInlineMixin};
    ${({ flexCenterAll }) => flexCenterAll && flexCenterAllMixin}
`;

const Box = ({ tag, children, ...props }: IBox) => (
    <BoxRoot tag={tag} {...props}>
        {children}
    </BoxRoot>
);

export default Box;
