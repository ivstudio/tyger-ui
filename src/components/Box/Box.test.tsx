import React from 'react';
import Box from './Box';
import { render } from '../../../test/utils/mockProvider';

describe('Box', () => {
    it('renders Box', () => {
        const { container } = render(
            <Box px="16" py="16">
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
                eiusmod tempor incididunt ut labore et dolore magna aliqua.
            </Box>
        );
        expect(container.firstChild).toMatchSnapshot();
    });
});
