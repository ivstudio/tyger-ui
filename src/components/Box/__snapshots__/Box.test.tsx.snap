// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box renders Box 1`] = `
<div
  class="sc-bczRLJ bJGtpP"
>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
</div>
`;
